src/domain/user.js

```diff
@@ -141,9 +141,9 @@ export default class User {
   static async findById(id, role) {
     const isTeacher = role === 'TEACHER'
     if (isTeacher) {
-      return _findByUniqueAsATeacher('id', id)
+      return _findByUniqueAsATeacher(id, role)
     }
-    return _findByUnique('id', id)
+    return User._findByUnique('id', id)
```

Comment:
this function should be without User class :
the correct one is ---->>  return _findByUnique('id', id)